src/data-sources/plugins/DataSourcePlugin.ts

```diff
@@ -116,6 +118,21 @@ export abstract class DataSourcePlugin<T> implements IDataSourcePlugin {
   }
 
   failure(error: any): void { 
+    ToastActions.addToast({ text: this.errorToMessage(error) });
     return error;
   }
+
+  private errorToMessage(error: any): string {
+    const exception = error as Error;
+    if (exception == null) {
+      return error;
```

Comment:
This was actually a misunderstanding on my part how `as` works in TypeScript. I assumed that it worked like in C# (try a cast and if it's not castable, return null). Fixed in 5961bc7 by using `instanceof`.